### PR TITLE
fix(logging): log rule name or file name with log entries

### LINKED ISSUE #313: Logging prefixes are now different
I am not sure exactly what changed, guessing it happened in #286  ? 

However, we can no longer set the log level for jruby using 'jsr223.jruby' as the names for some of the classes seemed to have changed.

I see logger names like: 'jsr223.jruby.OpenHAB::DSL::Rules::AutomationRule'

I am pretty sure that used to be 'jsr223.jruby.OpenHAB.DSL.:Rules.AutomationRule'

### PR DESCRIPTION
**Pull request by boc-tothefuture**
Closes #313 and #84

### CODE REVIEW

File: features/support/openhab.rb [hunk 1]
```diff
@@ -119,6 +119,13 @@ def check_log_regexp(regexp)
   File.foreach(openhab_log).grep(regexp).any?
 end
 
+def check_log_regex(entry)
```
Comment (jimtng) [inline features/support/openhab.rb:122]: there's `check_log_regexp` above. I suggest merging the two.

File: features/step_definitions/openhab.rb [hunk 2]
```diff
@@ -32,11 +32,26 @@
...
 
-# Then('It should log {string} within {int} seconds') do |string, seconds|
-#  wait_until(seconds: seconds, msg: "'#{string}' not found in log file (#{openhab_log}) within #{seconds} seconds") do
-#    check_log(string)
-#  end
-# end
+Then('It should log a line matching regex {string} within {int} seconds') do |regex, seconds|
+  wait_until(seconds: seconds.to_i,
+             msg: "'#{regex}' not found in log file (#{openhab_log}) within #{seconds} seconds") do
+    check_log_regexp(regex)
```
Comment (jimtng) [inline features/step_definitions/openhab.rb:38]: See the `check_log_regex` vs `check_log_regexp`

**Review by jimtng — COMMENTED** [review]
Great implementation! Could you add more tests for timers, and trigger delays?

**Comment by boc-tothefuture** [comment]
@jimtng fixed the duplicated log regex checks and added suggested tests.

File: lib/openhab/dsl/timers/timer.rb [hunk 3]
```diff
@@ -22,8 +23,11 @@ module Timers
       # @since 2.0.0
       class Timer < SimpleDelegator
         include OpenHAB::Log
+        include OpenHAB::Core::ThreadLocal
         extend Forwardable
 
+        def_delegator :@timer, :is_active, :active?
+        def_delegator :@timer, :is_running, :running?
```
[the remark this answers is not in the document]
Comment (boc-tothefuture) [reply lib/openhab/dsl/timers/timer.rb:30]: bad merge.

File: features/logging.feature [hunk 4]
```diff
@@ -29,32 +29,72 @@ Feature:  logging
...
         on_start
-        run { logger.info('Log Test') }
+        run { log_foo }
+      end
+
+      """
+    When I deploy the rules file named "log_file.rb"
+    Then It should log only "Foo" at level "INFO" from 'jsr223.jruby.log_file.log_test' within 5 seconds
+
+  Scenario: Logs in after blocks (timers) should have file and rule name in log prefix
```
[the remark this answers is not in the document]
Comment (boc-tothefuture) [reply features/logging.feature:70]: thanks!

File: docs/usage/misc/logging.md [hunk 5]
```diff
@@ -9,13 +9,25 @@ grand_parent: Usage
 
 # Logging
 
-Logging is available everywhere through the logger object.  The name of the rule file is automatically appended to the logger name. Pending [merge](https://github.com/openhab/openhab-core/pull/1885) into the core.
+Logging is available everywhere through the logger object. 
+
+Logging placed outside of a rule will have the name of the file appened to the logger name. The following entries are in a file named 'log_test'
```
Comment (jimtng) [inline docs/usage/misc/logging.md:14]: Sorry I didn't see this earlier.
```suggestion
Logging placed outside of a rule will have the name of the file appened to the logger name. The following entries are in a file named 'log_test.rb'
```

**Review by jimtng — APPROVED** [review]

**Pull request merged**